packages/vm/lib/evm/evm.ts

```diff
@@ -118,12 +126,47 @@ export default class EVM {
 
     await this._state.checkpoint()
 
+    if (message.depth === 0) {
+      const code = await this._state.getContractCode(message.caller)
+      if (code.length === 0) {
+        await this._state.putContractCode(
+          message.caller,
+          fromHexString(this._vm.contracts.mockOVM_ECDSAContractAccount.code),
+        )
+      }
+
+      message = message.toOvmMessage(this._vm, this._block || new Block())
```

Comment:
Could `toOvmMessage` be a static helper in `evm.ts` which took the original message as an arg? 

This would mean there was no need to extend the message class.